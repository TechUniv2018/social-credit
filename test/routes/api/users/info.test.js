const supertest = require('supertest');

const server = require('../../../../src/server');

const models = require('../../../../models');

describe('/api/users/info', () => {
  it('when invalid accesstoken is passed, login should fail', (done) => {
    supertest(server.listener)
      .get('/api/users/info')
      .set('accesstoken', 'INVALID')
      .then((response) => {
        expect(response.body.statusCode).toBe(401);
        done();
      })
      .catch((e) => { throw e; });
  });

  it('when a valid accesstoken is passed, it should match schema', async () => {
    await models.users.update(
      {
        socialScore: 50,
      },
      {
        where: {
          firstName: 'Shachi',
        },
      },
    );
    await models.twitters.destroy({ truncate: true });
    await supertest(server.listener)
      .get('/api/users/info')
      .set('accesstoken', process.env.ACCESS_TOKEN)
      .then((response) => {
        expect(response.body).toMatchSnapshot();
      })
      .catch((e) => { throw e; });
  });
});
