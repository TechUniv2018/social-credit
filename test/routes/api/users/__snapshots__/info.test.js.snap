// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/api/users/info when a valid accesstoken is passed, it should match schema 1`] = `
Object {
  "data": Object {
    "breakDown": Object {
      "facebook": Object {
        "friendsCount": 27,
        "impact": 5,
      },
      "twitter": Object {
        "followersCount": 0,
        "impact": 0,
        "isVerified": false,
        "secondFollowersCount": 0,
      },
    },
    "firstName": "Shachi",
    "lastName": "Mathur",
    "maxAmount": 500000,
    "socialScore": 50,
  },
  "statusCode": 200,
}
`;
